src/opencl_bitcracker_fmt_plug.c

```diff
@@ -0,0 +1,630 @@
+	int j, fileLen;
+	
+	if(!diskImage)
+		exit(1);
+
+	fseek(diskImage, 0, SEEK_END);
+    fileLen=ftell(diskImage);
+    fseek(diskImage, 0, SEEK_SET);
+	for (j=0;j<fileLen;j++){
+		c = fgetc(diskImage);
+		while ((unsigned char)c == signature[i]){
+			c = fgetc(diskImage);
+			i++;
+		}
+		if (i == 8) {
+			match=1;
+			printf("[BitCracker] -> Signature found at 0x%08lx\n", (ftell(diskImage)-i-1));
+			fseek(diskImage, 1, SEEK_CUR);
+			version = fgetc(diskImage);
+			printf("[BitCracker] -> Version: %d ", version);
+			if (version == 1)
+				printf("(Windows Vista)\n");
+			else if (version == 2)
+				printf("(Windows 7 or later)\n");
+			else{
+				printf("\nBitCracker] -> Invalid version, looking for a signature with valid version..\n");
+			}
+		}
+		i=0;
+		while ((unsigned char)c == vmk_entry[i]){
+			c = fgetc(diskImage);
+			i++;
+		}
+		if (i == 4){
+			printf("[BitCracker] -> VMK entry found at 0x%08lx\n", (ftell(diskImage)-i-3));
+			fseek(diskImage, 27, SEEK_CUR);
+			if (((unsigned char)fgetc(diskImage) == key_protection_type[0]) && ((unsigned char)fgetc(diskImage) == key_protection_type[1])) {
+				printf("[BitCracker] -> Key protector with user password found\n");
+				fseek(diskImage, 12, SEEK_CUR);
+				fillBuffer(diskImage, salt_bitcracker, 16);
+				printf("[BitCracker] -> Salt:");
+				cpu_print_hex(salt_bitcracker, 16);
+				fseek(diskImage, 83, SEEK_CUR);
+				if (((unsigned char)fgetc(diskImage) != value_type[0]) || ((unsigned char)fgetc(diskImage) != value_type[1])){
+					printf("[BitCracker] -> Error: VMK not encrypted with AES-CCM\n");
+					exit(1);
+				}
+				fseek(diskImage, 3, SEEK_CUR);
+				fillBuffer(diskImage, nonce, 12);
+				printf("[BitCracker] -> Nonce:");
+				cpu_print_hex(nonce, 12);
+				fillBuffer(diskImage, mac, 16);
+				printf("[BitCracker] -> MAC:");
+				cpu_print_hex(mac, 16);
+				fillBuffer(diskImage, encryptedVMK, 44);
+				printf("[BitCracker] -> Encrypted VMK:");
+				cpu_print_hex(encryptedVMK, 44);
+				break;
+			}
+		}
+		i=0;
+		
+	}
+	fclose(diskImage);
+	if (match==0){
+		printf("BitCracker] -> Error while extracting data: No signature found!\n");
+		exit(1);
+	}
+}
+
+static void set_key(char *key, int index)
+{
+	char tmp[BITCRACKER_MAX_INPUT_PASSWORD_LEN+2];
+	int size = strlen(key);
+
+	inbuffer_size[index] = size;
+	memset(tmp, 0, BITCRACKER_MAX_INPUT_PASSWORD_LEN+2);
+	memcpy(tmp, key, size);
+	if(size < 16) tmp[size] = 0x80;
+	memcpy((inbuffer+(index*BITCRACKER_MAX_INPUT_PASSWORD_LEN)), tmp, BITCRACKER_MAX_INPUT_PASSWORD_LEN);
+}
+
+static char *get_key(int index)
+{
+	static char ret[BITCRACKER_MAX_INPUT_PASSWORD_LEN + 1];
+	memcpy(ret, inbuffer+(index*BITCRACKER_MAX_INPUT_PASSWORD_LEN), inbuffer_size[index]);
+	return ret;
+}
+
+static int crypt_all(int *pcount, struct db_salt *salt)
+{
+	int count = *pcount;
+	int numPassword = count;
+	char fileNameAttack[] = "$JOHN/kernels/bitcracker_kernel.cl";
+   	char opt[1024];
+    unsigned int tmp_global, IV0, IV4, IV8, IV12;
```

Comment:
There seems to be inconsistent indentation here and also at different places in the code.